spmd/tensor/redistribute.py

```diff
@@ -76,20 +73,50 @@ def redistribute_spmd_tensor(
             elif current.is_replicate():
                 attempted_transforms.append(target)
                 # slice/narrow the tensor to corresponding local shard then return shard tensor
+                # TODO: support multi dim redistribute require to not use get_rank, as it would
+                # get global_rank and in the case of multidim mesh it's not correct!
                 new_local_tensor = local_tensor.narrow(
                     shard_dim, my_rank * chunk_size, chunk_size
                 )
             else:
                 # diff shard dim on new placement, record in attempted transforms
                 attempted_transforms.append(current)
+        elif target.is_partial():
+            if current.is_replicate():
+                if is_backward:
```

Comment:
I'm  trying to understand why this would be called in the first place in the backward. It thought this would only be called if it was necessary for the next operation to succeed, otherwise i thought we'd never do replicated to partial. Is there a case where this is going to hit even if it's not necessary?

Reply:
Also is this the only case where we will silently ignore the desired target?

Reply:
Yeah I think this should be the only case we silently ignore the desired target, and it should only happen in the backward of a manual call from user, it should not happen in operator dispatch